notes/connection-lifecycle.md

```diff
@@ -0,0 +1,77 @@
+# Connection lifecycle
+
+## Current implementation
+
+`HTTPConnection` should be instantiated with `host` and `port` of the
+**first origin being connected to** to reach the target origin. This either means
+the target origin itself or the proxy origin if one is desired.
+
+If the connection is an `HTTPSConnection`, then `HTTPSConnection.set_cert()` should
+be called once with information about how to verify certificates. (What is stopping us
+from putting this information in `HTTPSConnection.__init__()`?)
```

Comment:
Good question, I don't see any reason to do that. `set_cert` does not require an active connection and moving it into `__init__` would simplify the current code and let us get rid of `_prepare_conn`.